pkg/volume/gce_pd/gce_util.go

```diff
@@ -105,32 +106,41 @@ func (gceutil *GCEDiskUtil) CreateVolume(c *gcePersistentDiskProvisioner) (strin
 		case "zones":
 			zonesPresent = true
 			configuredZones = v
-		case "replica-zones":
-			replicaZonesPresent = true
-			configuredReplicaZones = v
+		case "replication-type":
+			replicationType = strings.ToLower(v)
 		case volume.VolumeParameterFSType:
 			fstype = v
 		default:
 			return "", 0, nil, "", fmt.Errorf("invalid option %q for volume plugin %s", k, c.plugin.GetPluginName())
 		}
 	}
 
-	if ((zonePresent || zonesPresent) && replicaZonesPresent) ||
-		(zonePresent && zonesPresent) {
-		// 011, 101, 111, 110
-		return "", 0, nil, "", fmt.Errorf("a combination of zone, zones, and replica-zones StorageClass parameters must not be used at the same time")
+	if zonePresent && zonesPresent {
+		return "", 0, nil, "", fmt.Errorf("the 'zone' and 'zones' StorageClass parameters must not be used at the same time")
+	}
+
+	if replicationType == replicationTypeRePD && zonePresent {
+		// If a user accidentally types 'zone' instead of 'zones', we want to throw an error instead of
+		// assuming that 'zones' is empty and proceed by randomly selecting zones.
+		return "", 0, nil, "", fmt.Errorf("the '%s' replication type does not support the 'zone' parameter; use 'zones' instead", replicationTypeRePD)
 	}
 
 	// TODO: implement PVC.Selector parsing
 	if c.options.PVC.Spec.Selector != nil {
 		return "", 0, nil, "", fmt.Errorf("claim.Spec.Selector is not supported for dynamic provisioning on GCE")
 	}
 
-	if !zonePresent && !zonesPresent && replicaZonesPresent {
-		// 001 - "replica-zones" specified
-		replicaZones, err := volumeutil.ZonesToSet(configuredReplicaZones)
-		if err != nil {
-			return "", 0, nil, "", err
+	switch replicationType {
+	case replicationTypeRePD:
+		var replicaZones sets.String
+		if configuredZones == "" {
+			// Parameter to ZonesToSet() can't be empty.
+			replicaZones = make(sets.String)
```

Comment:
Just do a `nil` check inside `createRegionalPD(...)`

Reply:
`ZonesToSet()` below can't take an empty string as input, so we still need this branch to set `replicaZones` to `nil`.

Reply:
Looks like `replicaZones` is only used inside `createRegionalPD(...)`. Constructing it outside is a little weird. How about passing in the string `configuredZones` to `createRegionalPD(...)` and then constructing `replicaZones` internally as needed?